### PR TITLE
DOC: whatsnew & linked edits

### PR DESCRIPTION
**Pull request by jreback**

### CODE REVIEW

**Comment by jreback** [comment]
small edits for consistency in the whatsnew; some incorrect links & section headings.

@datapythonista @TomAugspurger

File: doc/source/basics.rst [hunk 1]
```diff
@@ -64,37 +64,37 @@ NumPy's type system to add support for custom arrays
...
    s.index.array
 
 :attr:`~Series.array` will always be an :class:`~pandas.api.extensions.ExtensionArray`.
-The exact details of what an ``ExtensionArray`` is and why pandas uses them is a bit
+The exact details of what an :class:`~pandas.api.extensions.ExtensionArray` is and why pandas uses them is a bit
 beyond the scope of this introduction. See :ref:`basics.dtypes` for more.
 
 If you know you need a NumPy array, use :meth:`~Series.to_numpy`
-or :meth:`numpy.asarray`.
+or :meth:`np.asarray`.
```
Comment (datapythonista) [inline doc/source/basics.rst]: did you check if this renders correctly? I think we need to use `numpy` and not `np` here.

File: doc/source/basics.rst [hunk 2]
```diff
@@ -64,37 +64,37 @@ NumPy's type system to add support for custom arrays
...
+an :class:`~pandas.api.extensions.ExtensionArray`, :meth:`~Series.to_numpy`
 may involve copying data and coercing values. See :ref:`basics.dtypes` for more.
 
 :meth:`~Series.to_numpy` gives some control over the ``dtype`` of the
 resulting :class:`ndarray`. For example, consider datetimes with timezones.
 NumPy doesn't have a dtype to represent timezone-aware datetimes, so there
 are two possibly useful representations:
 
-1. An object-dtype :class:`ndarray` with :class:`Timestamp` objects, each
+1. An object-dtype :class:`np.ndarray` with :class:`Timestamp` objects, each
```
Comment (datapythonista) [inline doc/source/basics.rst]: same

File: doc/source/basics.rst [hunk 3]
```diff
@@ -64,37 +64,37 @@ NumPy's type system to add support for custom arrays
...
    with the correct ``tz``
-2. A ``datetime64[ns]`` -dtype :class:`ndarray`, where the values have
+2. A ``datetime64[ns]`` -dtype :class:`np.ndarray`, where the values have
```
Comment (datapythonista) [inline doc/source/basics.rst]: same

**Review by datapythonista — COMMENTED** [review]
looks good, just one thing I'm unsure

File: doc/source/basics.rst (hunk 1 above)
Comment (jreback) [reply doc/source/basics.rst]: doesn’t render now with numpy

Comment (datapythonista) [reply doc/source/basics.rst]: I see using `numpy` creating the links, for example: https://pandas-docs.github.io/pandas-docs-travis/basics.html#searchsorted

Comment (datapythonista) [reply doc/source/basics.rst]: That's weird, not sure why in this case it's not working. But if `np` works in this case, that's good enough for now (in the future would be good to understand what's going on, and make it work in all the cases for one of them).

Comment (jreback) [reply doc/source/basics.rst]: need numpy.ndarray.asarray i think

**Pull request merged**